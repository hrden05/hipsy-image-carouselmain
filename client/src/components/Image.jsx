import React from 'react';

class Image extends React.Component {
  constructor() {
    super();
    this.state = {
      poenis: [],
    };
  }

  render() {
    return (
      <div className="Image">
        <img src={this.props.source} alt={this.props.product_id} />
      </div>
    );
  }
}

export default Image;
