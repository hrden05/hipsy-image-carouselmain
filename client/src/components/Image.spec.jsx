import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Adapter from '../setupTests';
import Image from './Image';

describe('<Image />', () => {
  it('Should render correctly in "debug" mode', () => {
    const wrapper = shallow(<Image debug />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should mount in a full DOM', () => {
    const wrapper = mount(<Image />);
    expect(wrapper.find('.Image').length).toBe(2);
    wrapper.unmount();
  });
});
