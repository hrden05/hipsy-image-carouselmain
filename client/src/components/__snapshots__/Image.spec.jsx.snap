// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> Should render correctly in "debug" mode 1`] = `
<div
  className="Image"
>
  <div
    className="Image"
    style={
      Object {
        "alt": "undefined",
        "backgroundImage": "url(undefined)",
        "border": "1px solid white",
        "borderRadius": "4px",
      }
    }
  />
</div>
`;
